ts/model_loader.py

```diff
@@ -107,36 +107,21 @@ def load(
                     handler
                 )
             )
-        if function_name is None:
-            function_name = "handle"
-
-        if hasattr(module, function_name):
-            entry_point = getattr(module, function_name)
-            service = Service(
-                model_name,
-                model_dir,
-                manifest,
-                entry_point,
-                gpu_id,
-                batch_size,
-                limit_max_image_pixels,
-            )
-
-        envelope_class = None
-        if envelope is not None:
-            envelope_class = self._load_default_envelope(envelope)
-
         function_name = function_name or "handle"
+
         if hasattr(module, function_name):
             entry_point, initialize_fn = self._get_function_entry_point(
                 module, function_name
             )
         else:
             entry_point, initialize_fn = self._get_class_entry_point(module)
 
-        if envelope_class is not None:
-            envelope_instance = envelope_class(entry_point)
-            entry_point = envelope_instance.handle
+        if envelope is not None:
+
+            envelope_class = self._load_default_envelope(envelope)
+            if envelope_class is not None:
+                envelope_instance = envelope_class(entry_point)
+                entry_point = envelope_instance.handle
```

Comment:
It would be good to include a unit test in [test_model_loader.py](https://github.com/pytorch/serve/blob/master/ts/tests/unit_tests/test_model_loader.py) similar to `test_load_func_model` along with the `envelope` argument (potentially `JSONEnvelope` for example) and then validating that the `service._entry_point` is appropriately set to the envelope class `handle` method. This should help test the refactoring and also resolve the code coverage warning.

Reply:
I added a test called `test_load_model_with_envelope`.